lib/utils/restClient.cpp

```diff
@@ -71,16 +72,35 @@ void restClient::event_thread() {
         FATAL_ERROR_NON_OO("restClient: Failure creating new event_base.");
     }
 
+    // The event loop will run in this seperate thread. We have to schedule requests from
```

Comment:
I'm not going to pretend I've read the changes to this file in any detail. What would be good here would be to have much better overview documentation of the structure of this class, and why it needs to be like this. If you were to disappear, I can't imagine how we would maintain this one. Particularly given that there are subtle restrictions on what libevent itself can handle.